### ISSUE TITLE
flox has config to disable auto upgrade

### ISSUE THREAD

**Issue by ghudgins** [flox/flox#1308]
**When** I don't want Flox to update (eventual enterprise reasons? I like it how it is?)
**I need** to disable Flox's version check
**So I can** turn off the message that posts every so often in the CLI

### Acceptance Criteria

Write a `flox config` to disable the check

cc @ysndr

**Comment by sarendipitee** [comment]
This is much bigger an issue than just an annoying startup message.

If you use flox with agents, where each is spawning new shells constantly it's creating hundreds of lock contention and all but one shell gets a `⚠️  Waiting for another activation to complete (blocked by PID 47835)...` hang creating long delays in shell startup.

**Comment by mkenigs** [comment]
I'm not sure if `⚠️  Waiting for another activation to complete (blocked by PID 47835)...` has anything to do with auto-update, could that be a separate issue?

Also how long are the delays? We poll every 200ms to check if the other activation has completed, are you seeing delays longer than that after the first activation has completed? Do you have a `hook.on-activate` that takes a while?

**Comment by sarendipitee** [comment]
@mkenigs Negative. [Nothing in on-activate.](https://github.com/sarendipitee/dotfiles/blob/3877960d73d6ac794984236e87521dc7316d6013/packages/flox/global-env/.flox/env/manifest.toml#L168-L179)

But I compiled flox from source on my [disable-update-checks branch](https://github.com/sarendipitee/flox/tree/disable-update-checks) and it hasn't occurred since, which is a pretty big smoking gun I'd say.

**Comment by mkenigs** [comment]
Interesting, I think we probably need to dig in a bit more to understand root cause there. We don't want update checks blocking activate when they're enabled. Do you have a more focused reproducer?

**Comment by sarendipitee** [comment]
To reproduce, I would ask an LLM in a harness that supports parallel tool calls to spawn several Shells tools at once, each running a simple command like `ls`.

I think you may be right that it wasn't the update check that was blocking.

I goofed and my update check opt-out branch was branched from [fix-activation-backoff-retry](https://github.com/flox/flox/pull/4241). I think that branch is actually the real fix for the problem I was observing (or possibly the combination of both!)
